sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/V1WriteCommandSuite.scala

```diff
@@ -93,9 +93,10 @@ trait V1WriteCommandSuiteBase extends SQLTestUtils {
     assert(optimizedPlan != null)
     // Check whether exists a logical sort node of the write query.
     // If user specified sort matches required ordering, the sort node may not at the top of query.
-    assert(optimizedPlan.exists(_.isInstanceOf[Sort]) == hasLogicalSort,
-      s"Expect hasLogicalSort: $hasLogicalSort," +
-        s"Actual: ${optimizedPlan.exists(_.isInstanceOf[Sort])}")
+    def isSort(plan: LogicalPlan): Boolean =
+      plan.isInstanceOf[Sort] || plan.isInstanceOf[Clustering]
```

Comment:
do most of the test cases expect `Clustering` now?

Reply:
Only the tests that enable V1 writes and the output ordering not matching the required ordering will contain `Clustering`.